src/transform/src/analysis/equivalences.rs

```diff
@@ -297,30 +297,39 @@ pub struct EquivalenceClasses {
 }
 
 impl EquivalenceClasses {
+    /// Comparator function for the complexity of scalar expressions. Can be used when we need to
+    /// decide which of several equivalent expressions to use.
+    pub fn mir_scalar_expr_complexity(
+        e1: &MirScalarExpr,
+        e2: &MirScalarExpr,
+    ) -> std::cmp::Ordering {
+        match (e1, e2) {
+            (MirScalarExpr::Literal(_, _), MirScalarExpr::Literal(_, _)) => e1.cmp(e2),
+            (MirScalarExpr::Literal(_, _), _) => std::cmp::Ordering::Less,
+            (_, MirScalarExpr::Literal(_, _)) => std::cmp::Ordering::Greater,
+            (MirScalarExpr::Column(_), MirScalarExpr::Column(_)) => e1.cmp(e2),
+            (MirScalarExpr::Column(_), _) => std::cmp::Ordering::Less,
+            (_, MirScalarExpr::Column(_)) => std::cmp::Ordering::Greater,
+            (x, y) => {
+                // General expressions should be ordered by their size,
+                // to ensure we only simplify expressions by substitution.
+                let x_size = x.size();
```

Comment:
You could write this as a match over `x.size().cmp(&y.size())` if I'm not mistaken, but whatever seems simpler to you!